### PR TITLE
Fix list command to work with grouping.

### PR DESCRIPTION
**Pull request by rnixx**
Fix list command to work with grouping. Always returns list of dictionaries now. Make sure to adopt your code since this is an API change.

Maybe we should add an environment variable which can be used to force the old behavior?

### CODE REVIEW

**Comment by Mic92** [comment]
While at it, I added some stuff on top. Ping me, if you are ready.

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,6 @@ matrix:
     - python: nightly
 
 install:
-  - if [[ $TRAVIS_PYTHON_VERSION != '3.2' ]]; then pip install Twisted; fi
+  - if [[ $TRAVIS_PYTHON_VERSION != '3.4' ]]; then pip install Twisted; fi
```
Comment (Mic92) [inline .travis.yml]: This can go than?

Comment (rnixx) [reply .travis.yml]: yes

**Comment by rnixx** [comment]
Should we add a B/C control flag? Or just let things break? Apart from this i'm done so far.

**Comment by Mic92** [comment]
I would not add an compatibility flag, but I would add a simple example showing,
what the return type looks now:
 

```
>>> client.list('album', 'artist')
[{'album': '20th_Century_Masters_The_Millenium_Collection', 'artist': 'Eric Clapton'}, 
 {'album': "Aerosmith's Greatest Hits", 'artist': 'Aerosmith'}]
```

**Comment by rnixx** [comment]
Where to add the docs?
The commands reference is generated and describes no return values at all.
The examples contain code to try, so some may just adjust the helloworld example.
And the changelog not contains code either.
Then better leave as is.

**Pull request merged**